docs/README.md

```diff
@@ -0,0 +1,327 @@
+
+- [ARCOptimization.rst](/docs/ARCOptimization.rst):
+  Covers how ARC optimization works, with several examples.
+  TODO: Not clear if this is intended to be a topic guide
+  or a reference guide.
+- [CompilerPerformance.md](/docs/CompilerPerformance.md):
+  Thoroughly discusses different ways of measuring compiler performance
+  and common pitfalls.
+  TODO: Consider breaking up into one high-level
+  topic guide explaining key concepts and individual how-to guides that
+  can be expanded independently.
+  Also, it's not the right place to explain details of different compiler modes.
+- [DevelopmentTips.md](/docs/DevelopmentTips.md):
+  Contains an assortment of tips for better productivity when working on the
+  compiler.
+  TODO: Might be worthwhile to make a dedicated how-to guide or topic guide.
+  It might also be valuable to introduce the tips in context, and have the
+  topic guide link to all the different tips.
+- [Diagnostics.md](/docs/Diagnostics.md):
+  Describes how to write diagnostic messages and associated educational notes.
+  TODO: Consider splitting into how-tos (e.g. for educational notes)
+  and recommended practices (on writing diagnostics).
+- [HowSwiftImportsCAPIs.md](/docs/HowSwiftImportsCAPIs.md):
+  Contains a thorough description of the mapping between
+  C/ObjC entities and Swift entities.
+  TODO: Not clear if this is intended to be language documentation
+  (for Swift developers), a topic guide or a reference guide.
+- [OptimizerCountersAnalysis.md](/docs/OptimizerCountersAnalysis.md):
+  TODO: Consider breaking up into a how-to guide
+  on dumping and analyzing the counters
+  and a topic guide that gives an overview
+  of the the counter collection system.
+- [Testing.md](/docs/Testing.md):
+  TODO: Consider splitting into a how-to guide on writing a new test case
+  and a topic guide on how the compiler is tested.
+- [SwiftIndent.md](/docs/SwiftIndent.md):
+  TODO: Unclear if this is intended to be a topic guide or a reference guide.
+- [Random.md](/docs/Random.md): Stub.
+
+### Archive
+
+- [FailableInitializers.rst](/docs/FailableInitializers.rst):
+  Superseded by documentation in [The Swift Programming Language][].
+- [InitializerProblems.rst](/docs/InitializerProblems.rst):
+  Describes some complexities around initialization in Swift.
+  TODO: It would be great to have a topic guide say `InitializationModel.md`
+  that covers the initialization model and new attributes like
+  `@_hasMissingDesignatedInitializers`.
+  Some of this is covered in
+  [TSPL's initialization section][]
+  but that doesn't include newly added attributes.
+- [Modules.rst](/docs/Module.rst): for Swift pre-1.0.
+- [Swift3Compatibility.md](/docs/Swift3Compatibility.md):
+  Discusses the Swift 3 -> Swift 4 migration.
+- [StoredAndComputedVariables.rst](): for Swift pre-1.0.
+
+[TSPL's initialization section]: https://docs.swift.org/swift-book/LanguageGuide/Initialization.html
+<!-- Ubuntu14.md: We should delete this. -->
```

Comment:
This can be slotted under "How-To Guides"; it's explicitly referenced from the main README.

Reply:
>  it's explicitly referenced from the main README.

Yes, that reference in the README needs to be deleted too. 16.04 LTS, 18.04 LTS and now 20.04 LTS should work. I really should've submitted that PR first; let me do that now.

Reply:
https://github.com/apple/swift/pull/32010